File: backend/src/main/java/com/library/backend/entity/Book.java
package com.library.backend.entity;

import com.baomidou.mybatisplus.annotation.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("book")
public class Book {
    @TableId(type = IdType.AUTO)
    private Long id;

    @NotBlank(message = "书名不能为空")
    private String title;

    @NotBlank(message = "作者不能为空")
    private String author;

    @NotBlank(message = "ISBN不能为空")
    private String isbn;

    private String category;

    @NotNull(message = "价格不能为空")
    @Min(value = 0, message = "价格不能小于0")
    private BigDecimal price;

    @NotNull(message = "库存不能为空")
    @Min(value = 0, message = "库存不能小于0")
    private Integer stock;

    private String description;
    private String cover;
    
    @TableLogic
    private Integer deleted;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/book.ts
import request from '../utils/request'

export interface Book {
  id?: number
  title: string
  author: string
  isbn: string
  category: string
  price: number
  stock: number
  description: string
  createTime?: string
  updateTime?: string
}

export function getBookList(params: any) {
  return request({
    url: '/books',
    method: 'get',
    params
  })
}

export function getBook(id: number) {
  return request({
    url: `/books/${id}`,
    method: 'get'
  })
}

export function createBook(data: Book) {
  return request({
    url: '/books',
    method: 'post',
    data
  })
}

export function updateBook(data: Book) {
  return request({
    url: '/books',
    method: 'put',
    data
  })
}

export function deleteBook(id: number) {
  return request({
    url: `/books/${id}`,
    method: 'delete'
  })
}
